### PR TITLE
User Interaction Extraction

### PR DESCRIPTION
**Pull request by wcjordan**
**What it accomplishes:**

* Identify and extract click, input, and scroll events (`sources` 2, 5, 3) into a list of `UserInteraction` records.

**How to verify:**

* Supply mixed-source events to the extractor and verify only the intended interactions are returned, with correct fields populated.

--
Introduces a new extractor to convert rrweb incremental snapshot events into structured UserInteraction records, adds extensive tests for click, input, and scroll cases (including edge conditions), and updates the feature‐extraction TODO documentation.

* Implement extract_user_interactions (with click, input, scroll helpers) and import UserInteraction
* Add comprehensive unit tests for filtering, field mapping, order preservation, and missing‐field handling
* Update todo.md to mark user interaction extraction tasks as complete

### CODE REVIEW

File: test_gen/feature_extraction/extractors.py [hunk 1]
```diff
@@ -128,3 +128,107 @@ def _extract_node_removals(data: dict, timestamp: int) -> List[DomMutation]:
...
+
+    Note:
+        Events with other source values or missing required fields are ignored.
+        The function preserves event order in the returned list.
+    """
+    interactions = []
+
+    for event in events:
+        # Only process IncrementalSnapshot events
+        if event.get("type") != 3:
```
[the remark this answers is not in the document]
Comment (wcjordan) [reply test_gen/feature_extraction/extractors.py:161]: Added a future todo
